examples/real_world_scenario.rs

```diff
@@ -0,0 +1,60 @@
+extern crate thread_profiler;
+
+#[cfg(feature = "thread_profiler")]
+use thread_profiler::ProfileScope;
+
+use std::thread;
+use std::string::String;
+use std::vec::Vec;
+
+fn main() {
+	if !cfg!(feature = "thread_profiler") {
+		panic!("This example must be run with the 'thread_profiler' feature enabled");
+	}
+	let workers = 10;
+	let mut tasks = Vec::new();
+	for i in 0..workers {
+		tasks.push(thread::spawn(move || {
+			#[cfg(feature = "thread_profiler")]
+			thread_profiler::register_thread_with_profiler();
+			#[cfg(feature = "thread_profiler")]
+			let _p = ProfileScope::new(format!("worker: thread_{}", i));
+			let result = {
+				#[cfg(feature = "thread_profiler")]
+				ProfileScope::new(format!("worker: thread_{}::calculation", i));
```

Comment:
any reason this is not using the macro?

Reply:
Update this so that it has an example using the macro and a description of the advantages as well as a description of using `ProfileScope::new()` and the advantages of using that.